lib/icinga/checkable-check.cpp

```diff
@@ -242,6 +242,16 @@ void Checkable::ProcessCheckResult(const CheckResult::Ptr& cr, const MessageOrig
 			OnReachabilityChanged(this, cr, children, origin);
 	}
 
+	if (recovery) {
+		// W/o this children could recover too late, after their problem notifications have been sent.
+		for (auto& child : children) {
+			if (child->GetProblem() && child->GetEnableActiveChecks() && !child->IsLikelyToBeCheckedSoon()) {
+				ObjectLock oLock (child);
+				child->SetNextCheck(now + 10);
```

Comment:
Why `+ 10`?

Reply:
Just to be sure we have enough time to finish processing the current check result.